doc/api/esm.md

```diff
@@ -313,50 +277,193 @@ If a package has no exports, setting `"exports": false` can be used instead of
 `"exports": {}` to indicate the package does not intend for submodules to be
 exposed.
 
-Exports can also be used to map the main entry point of a package:
+Any invalid exports entries will be ignored. This includes exports not
+starting with `"./"` or a missing trailing `"/"` for directory exports.
+
+Array fallback support is provided for exports, similarly to import maps
+in order to be forward-compatible with fallback workflows in future:
 
 <!-- eslint-skip -->
 ```js
-// ./node_modules/es-module-package/package.json
 {
   "exports": {
-    ".": "./main.js"
+    "./submodule": ["not:valid", "./submodule.js"]
   }
 }
 ```
 
-where the "." indicates loading the package without any subpath. Exports will
-always override any existing `"main"` value for both CommonJS and
-ES module packages.
+Since `"not:valid"` is not a supported target, `"./submodule.js"` is used
+instead as the fallback, as if it were the only target.
+
+Defining a `"."` export will define the main entry point for the package,
+and will always take precedence over the `"main"` field in the `package.json`.
 
-For packages with only a main entry point, an `"exports"` value of just
-a string is also supported:
+This allows defining a different entry point for Node.js versions that support
+ECMAScript modules and versions that don't, for example:
 
 <!-- eslint-skip -->
 ```js
-// ./node_modules/es-module-package/package.json
 {
-  "exports": "./main.js"
+  "main": "./main-legacy.cjs",
+  "exports": {
+    ".": "./main-modern.mjs"
+  }
```

Comment:
I think this suggestion creates an issue where `require` works in node.js <13 but fails in node.js 13.  Assume that a pure CJS module supports node.js 8+ and `require`'s this module.  In node.js 8-12 it will successfully require `./main-legacy.cjs' but in node.js 13 it will try to require `./main-modern.mjs`.  For this to work correctly you would need `exports['.'].require` to either point at `./main-legacy.cjs` or if desired a `./main-modern.cjs`.

Reply:
Yes, this suggestion is not a comprehensive workflow, but it is intended to be only a technical example of how the field itself works. The practical advice is given in the publishing dual packages section further down. Agreed though we shouldn't have examples that are going to cause issues - I've updated both to be CJS as suggested.